### ISSUE TITLE
How to ignore E501: Line too long in docstrings

### ISSUE THREAD

**Issue by DanielNoord** [astral-sh/ruff#9200]
```python
def func():
    """
    My very long docstring that I can't seem to be able to get under 100 characters but that I don't want to change.
    """

    string = "My very long string that I can't seem to be able to get under 100 characters but that should be changed"
```
Running `ruff --select=E501 --line-length=100 test.py`.

How can I ignore E501 on the line in the docstring but not in the `string` line. I have tried to do this but can't figure out how to do so. `pylint` read the ignore comment even if it is in a docstring, would that be a possibility here?

**Comment by charliermarsh** [comment]
The `# noqa` for multi-line strings need to come at the end of the multi-line string, kinda like this:

```python
def func():
    """
    My very long docstring that I can't seem to be able to get under 100 characters but that I don't want to change.
    """  # noqa: E501
```

This ensures that the `# noqa` doesn't appear within the actual string contents.

**Comment by DanielNoord** [comment]
Thanks @charliermarsh, that's what I was looking for.

Not sure if these should be added to the `line-too-long` docs. Multi-line strings and docstrings are different objects and I couldn't find anything through google/GIthub when looking for docstrings. Adding a short example for them might help with discoverability.

**Comment by charliermarsh** [comment]
Good call -- it's mentioned in the docs [here](https://docs.astral.sh/ruff/linter/#error-suppression) but it should probably be mentioned in the rule documentation since this does come up often.

**Comment by 1vecera** [comment]
I want to suggest that E501 would not apply for docstings at all as there is [doc-line-too-long (W505)](https://docs.astral.sh/ruff/rules/doc-line-too-long/#doc-line-too-long-w505) for people that want to length check their docstrings and the overlap causes issues for people who are fine with long dosctrings, but want short code. Having noqa everywhere is ugly.

Should I open a new issue for that?

**Comment by MichaReiser** [comment]
@1vecera see https://github.com/astral-sh/ruff/issues/16577

**Comment by 1vecera** [comment]
thanks!
